Tools/clinic/libclinic/converter.py

```diff
@@ -0,0 +1,534 @@
+            self.use_converter()
+            return self.format_code("""
+                if (!{converter}({argname}, &{paramname})) {{{{
+                    goto exit;
+                }}}}
+                """,
+                argname=argname,
+                converter=self.converter)
+        if self.format_unit == 'O!':
+            cast = '(%s)' % self.type if self.type != 'PyObject *' else ''
+            if self.subclass_of in type_checks:
+                typecheck, typename = type_checks[self.subclass_of]
+                return self.format_code("""
+                    if (!{typecheck}({argname})) {{{{
+                        {bad_argument}
+                        goto exit;
+                    }}}}
+                    {paramname} = {cast}{argname};
+                    """,
+                    argname=argname,
+                    bad_argument=self.bad_argument(displayname, typename, limited_capi=limited_capi),
+                    typecheck=typecheck, typename=typename, cast=cast)
+            return self.format_code("""
+                if (!PyObject_TypeCheck({argname}, {subclass_of})) {{{{
+                    {bad_argument}
+                    goto exit;
+                }}}}
+                {paramname} = {cast}{argname};
+                """,
+                argname=argname,
+                bad_argument=self.bad_argument(displayname, '({subclass_of})->tp_name',
+                                               expected_literal=False, limited_capi=limited_capi),
+                subclass_of=self.subclass_of, cast=cast)
+        if self.format_unit == 'O':
+            cast = '(%s)' % self.type if self.type != 'PyObject *' else ''
+            return self.format_code("""
+                {paramname} = {cast}{argname};
+                """,
+                argname=argname, cast=cast)
+        return None
+
+    def set_template_dict(self, template_dict: TemplateDict) -> None:
+        pass
+
+    @property
+    def parser_name(self) -> str:
+        if self.name in libclinic.CLINIC_PREFIXED_ARGS: # bpo-39741
+            return libclinic.CLINIC_PREFIX + self.name
+        else:
+            return self.name
+
+    def add_include(self, name: str, reason: str,
+                    *, condition: str | None = None) -> None:
+        include = Include(name, reason, condition)
+        self.includes.append(include)
+
+
+ConverterType = Callable[..., CConverter]
+ConverterDict = dict[str, ConverterType]
+
+# maps strings to callables.
+# these callables must be of the form:
+#   def foo(name, default, *, ...)
+# The callable may have any number of keyword-only parameters.
+# The callable must return a CConverter object.
+# The callable should not call builtins.print.
+converters: ConverterDict = {}
+
+# maps strings to callables.
+# these callables follow the same rules as those for "converters" above.
+# note however that they will never be called with keyword-only parameters.
+legacy_converters: ConverterDict = {}
```

Comment:
These belong to the Argument Clinic _app_ (or CLI if you will), so IMO they should not be part of the Argument Clinic *library*. IMO, it is part of the Argument Clinic *app* config.

Reply:
I would prefer to move as much as possible inside libclinic, including the CLI and "app" logic. Apparently, we have a disagreement on that.

Reply:
I do not oppose to move the CLI / app into libclinic. What I'm saying is I think the global `converters` and `legacy_converters` dicts should be a part of the CLI config.

Reply:
Ok, that makes sense. The practical issue is to reorganize code (move code around) when they are inter-dependenies and most code is left in `clinic.py`. Code from libclinic/ cannot get code from `clinic.py`. Once more code will be moved to libclinic, it will be easier to group related code.